### PR TITLE
Don't require an app.json config file to start a native project

### PR DESCRIPTION
**Pull request by EvanBacon**
## Why

- https://github.com/expo/expo-cli/issues/1101
- parity with web

## How

Disable config existence validation in the `readConfigJsonAsync` & `readConfigJson` methods. 
- This should only apply to starting a project. Publishing, deploying, and building should all require an `app.json` to exist (for now).

## Test Plan

Run `expod start --ios` in the `packages/webpack-config/tests/minimum` project.

## Future 

- prompting for bundleID, package name, slug, etc... writes just that single value to a config (and creates it if it doesn't exist)
- building, publishing, deploying without an `app.json`.
- ability to create an `app.config.js` with a method that provides the default config for users to mix features into.
- restructure docs to inform you to create an `app.json` before editing.
- remove `app.json` from template blank.

### CODE REVIEW

File: packages/config/src/Config.ts [hunk 1]
```diff
@@ -489,11 +489,7 @@ function getRootPackageJsonPath(projectRoot: string, exp: ExpoConfig): string {
   return packageJsonPath;
 }
 
-export function readConfigJson(
-  projectRoot: string,
-  skipValidation: boolean = false,
-  skipNativeValidation: boolean = false
-): ProjectConfig {
+export function readConfigJson(projectRoot: string, skipValidation: boolean = true): ProjectConfig {
```
Comment (fson) [inline packages/config/src/Config.ts]:
```suggestion
export function readConfigJson(projectRoot: string, { skipValidation = true }: { skipValidation?: boolean } = {}): ProjectConfig {
```
We should get rid of these positional boolean arguments.
```
readConfigJson(projectRoot, true);  // what does true mean here? Need to look at the definition.
readConfigJson(projectRoot, { skipValidation: true });  // obvious at a glance
```

Comment (fson) [reply packages/config/src/Config.ts]: Also `skipValidation` sounds like doing validation is the default and you can optionally skip it, but the default now is `skipValidation = true`. Should it be `false` instead? Or rename the option to `requireExpoSDKVersion`?

File: packages/config/src/__tests__/Config-test.js [hunk 2]
```diff
@@ -141,23 +141,25 @@ describe('readConfigJson', () => {
...
-      const { exp, pkg } = readConfigJson('/no-package', true, true);
+      const { exp, pkg } = readConfigJson('/no-package');
       expect(exp.name).toBe(pkg.name);
       expect(exp.description).toBe(pkg.description);
     });
 
     it(`will throw if the app.json is missing`, () => {
-      expect(() => readConfigJson('/no-config')).toThrow(/app.json must include a JSON object./);
+      expect(() => readConfigJson('/no-config', { requireLocalConfig: true })).toThrow(
+        /app.json must include a JSON object./
```
Comment (fson) [inline packages/config/src/__tests__/Config-test.js]: I think this error message should include a bit more detail, maybe an example of a valid app.json and/or link to the relevant documentation.

**Comment by EvanBacon** [comment]
I think we need a new approach to this PR. Currently if a project is missing an `app.json` then we consider it to be a different kind of project when building in turtle. Seems like we should slowly deprecate the method in favor of a differently named method so we can work through each instance individually.

**Pull request closed without merging**